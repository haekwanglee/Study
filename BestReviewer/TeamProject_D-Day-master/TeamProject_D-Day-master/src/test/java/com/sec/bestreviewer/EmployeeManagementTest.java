package com.sec.bestreviewer;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;


import java.io.File;
import java.io.FileOutputStream;
import java.io.OutputStream;
import java.text.Format;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class EmployeeManagementTest {
    @Test
    public void testArgumentsEmptyInputFile() throws Exception {
        String[] args = {"./src/test/java/com/sec/bestreviewer/empty.txt", "./src/test/java/com/sec/bestreviewer/output.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();
        employeeManagement.run(args);

        File file = new File("./src/test/java/com/sec/bestreviewer/output.txt");

        assertTrue(file.exists());
    }

    @Test
    public void testArgumentsWrongArgsCount() throws Exception {
        String[] args = {"input.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            employeeManagement.run(args);
        });
    }

    @Test
    public void testArgumentsNotExistInputFile() throws Exception {
        String[] args = {"notexist.txt", "output.txt"};

        EmployeeManagement employeeManagement = new EmployeeManagement();

        Assertions.assertThrows(IllegalArgumentException.class, () -> {
            employeeManagement.run(args);
        });
    }

    @Test
    public void integrationTest() throws Exception {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_input.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }

    @Test
    public void integrationModifyTest() throws Exception {
        final String outputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_modify_output.txt";
        final String inputFileName = "./src/test/java/com/sec/bestreviewer/integration_test_modify_input.txt";

        String[] args = {inputFileName, outputFileName};
        File outputFile = new File(outputFileName);

        EmployeeManagement employeeManagement = new EmployeeManagement();

        employeeManagement.run(args);

        Approvals.verify(outputFile);
    }
}
