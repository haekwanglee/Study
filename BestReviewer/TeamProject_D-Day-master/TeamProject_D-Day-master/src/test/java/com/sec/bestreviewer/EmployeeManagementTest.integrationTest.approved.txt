SCH,9
SCH,NONE
SCH,02712462,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
SCH,09752663,VWCGM OH,CL3,010-6288-1811,19860720,ADV
SCH,94526445,VLPDHI LEE,CL4,010-7975-9766,19721111,PRO
SCH,NONE
SCH,NONE
SCH,NONE
SCH,99398134,JYFMBMT OH,CL4,010-5269-9657,19730321,ADV
SCH,NONE
SCH,02712462,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
SCH,NONE
SCH,NONE
SCH,NONE
SCH,02712462,UAFBOWU HONG,CL3,010-9054-6560,19780825,ADV
SCH,03717863,NSESFBLRK KIM,CL3,010-4108-4007,19770502,PRO
SCH,7
SCH,NONE
SCH,93728322,QSJYRB HONG,CL4,010-3602-4064,19690603,ADV
SCH,03249872,TYLNXWQN OH,CL3,010-7910-1669,19820828,ADV
